lib/src/file_watcher/polling.dart

```diff
@@ -73,17 +78,19 @@ class _PollingFileWatcher implements FileWatcher, ManuallyClosedWatcher {
     }
     if (_eventsController.isClosed) return;
```

Comment:
Is this relying on any sync-async behavior here? Should we add a return at the end of the catch to remove that?

Reply:
I'm not sure I understand

Reply:
I am also wondering if in the catch we should complete `ready` with this error if it hasn't completed yet? Otherwise it might be left hanging.

Reply:
sync-async is where all the synchronous code at the top of an async method actually runs synchronously (up to the first await) - possibly making it such that `_eventsController.isClosed` is true here immediately after the first error?

But, I just realized we are awaiting `close()` so actually it should be fine :).

Reply:
My other comment about ensuring `ready` gets completed (probably with an error) if it has not been completed yet, in the catch, is still relevant though :)

Reply:
> I am also wondering if in the catch we should complete ready with this error if it hasn't completed yet?

We never usually complete `ready` with an error, only ever success. I've added code to mark as ready in a few additional cases to reduce the chance we ever _don't_ do it. LMK if this seems right!

Reply:
I think it is fine, the error is still added to the event stream